packages/core/admin/ee/admin/src/pages/SettingsPage/pages/AuditLogs/components/TableRows.tsx

```diff
@@ -3,17 +3,38 @@ import React from 'react';
 import { Flex, IconButton, Tbody, Td, Tr, Typography } from '@strapi/design-system';
 import { onRowClick, stopPropagation } from '@strapi/helper-plugin';
 import { Eye } from '@strapi/icons';
+import { Entity } from '@strapi/types';
 import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
 
-import useFormatTimeStamp from '../hooks/useFormatTimeStamp';
+import { AuditLog } from '../../../../../../../../shared/contracts/audit-logs';
+import { useFormatTimeStamp } from '../hooks/useFormatTimeStamp';
 import { getDefaultMessage } from '../utils/getActionTypesDefaultMessages';
 
-const TableRows = ({ headers, rows, onOpenModal }) => {
+export type TableHeader = {
+  key: string;
+  name: keyof AuditLog;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  cellFormatter?: (cellValue: any) => string;
+  metadatas: {
+    label: string;
+    sortable: boolean;
+  };
+};
+
+type TableRowsProps = {
+  headers: TableHeader[];
+  rows: AuditLog[];
+  onOpenModal: (id: Entity.ID) => void;
+};
+
+export const TableRows = ({ headers, rows, onOpenModal }: TableRowsProps) => {
   const { formatMessage } = useIntl();
   const formatTimeStamp = useFormatTimeStamp();
 
-  const getCellValue = ({ type, value, model }) => {
+  // Not sure that 'value' can be typed properly
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Couldn't we say this has to be one of all available Attribute types of a content-type? Since there is the REST API in between this can probably be reduced to all JSON primitives, which they have somewhere in the BE code available already as type.

Reply:
Sorry, it is not clear to me, why would this be Attribute types 🤔 ? The auditlogs TableRow value seems to handle only AuditLogs attributes. I've already tried to use `value: AuditLog[keyof AuditLog];` but it leads to several changes. 
This type `AuditLog[keyof AuditLog];` doesn't seems to work properly because of the extends Pick<Entity, 'id'>. I can't understand why the AuditLog[keyof AuditLog] won't take into account the string type that exist in the AuditLog type 🤯 
![Screenshot 2023-12-11 at 17 09 26](https://github.com/strapi/strapi/assets/7756284/2b234928-cde7-41ac-a14d-94455d5e3538)
![Screenshot 2023-12-11 at 17 09 35](https://github.com/strapi/strapi/assets/7756284/1e2c231a-ee01-452e-8704-1d43c791b3be)

TBH I spend too much time on this, if you have any solution here, I'll take it.